config/ompi_setup_prrte.m4

```diff
@@ -16,130 +16,153 @@
 #                         reserved.
 # Copyright (c) 2009      Oak Ridge National Labs.  All rights reserved.
 # Copyright (c) 2019-2020 Intel, Inc.  All rights reserved.
+# Copyright (c) 2020      Amazon.com, Inc. or its affiliates.
+#                         All Rights reserved.
 # $COPYRIGHT$
 #
 # Additional copyrights may follow
 #
 # $HEADER$
 #
 
+# Check for / configure PRRTE package.  Unlike the other 3rd party
+# packages, prrte is either internal or not used (since prrte is a
+# pmix server, we don't need to link against it or such in the
+# external case).  So we don't have to do some of the complex steps we
+# do for pmix or hwloc in terms of external hunting.
+#
+# This macro will not change the environment.
+#
+# A Makefile conditional OMPI_WANT_PRRTE will be defined based on the
+# results of the build.  Unless --disable-internal-rte is specified,
+# the top-level configure will abort if the PRRTE configure fails.
 AC_DEFUN([OMPI_SETUP_PRRTE],[
-    OPAL_VAR_SCOPE_PUSH([opal_prrte_save_CPPFLAGS opal_prrte_save_CFLAGS opal_prrte_save_LDFLAGS opal_prrte_save_LIBS opal_prrte_args opal_prrte_save_enable_dlopen opal_prrte_save_enable_mca_dso opal_prrte_save_enable_mca_static opal_prrte_extra_libs opal_prrte_extra_ltlibs opal_prrte_extra_ldflags opal_prrte_save_with_libevent opal_prrte_save_with_hwloc opal_prrte_save_with_pmix])
-
-    opal_prrte_save_CFLAGS=$CFLAGS
-    opal_prrte_save_CPPFLAGS=$CPPFLAGS
-    opal_prrte_save_LDFLAGS=$LDFLAGS
-    opal_prrte_save_LIBS=$LIBS
-    opal_prrte_save_enable_dlopen=enable_dlopen
-    opal_prrte_save_enable_mca_dso=enable_mca_dso
-    opal_prrte_save_enable_mca_static=enable_mca_static
-    opal_prrte_save_with_libevent=with_libevent
-    opal_prrte_save_with_hwloc=with_hwloc
-    opal_prrte_save_with_pmix=with_pmix
+    OPAL_VAR_SCOPE_PUSH([internal_prrte_build])
+
+    internal_prrte_build=0
+    m4_ifdef([package_prrte],
+             [OMPI_SETUP_PRRTE_INTERNAL([internal_prrte_build=1], [internal_prrte_build=0])])
+
+    AM_CONDITIONAL([OMPI_WANT_PRRTE], [test "$internal_prrte_build" = "1"])
+
+    OPAL_VAR_SCOPE_POP
+])
+
+AC_DEFUN([OMPI_SETUP_PRRTE_INTERNAL], [
+    OPAL_VAR_SCOPE_PUSH([internal_prrte_args internal_prrte_extra_libs internal_prrte_happy deprecated_prefix_by_default print_prrte_warning])
+
+    opal_show_subtitle "Configuring PRRTE"
 
     AC_ARG_ENABLE([internal-rte],
                   [AC_HELP_STRING([--enable-internal-rte],
                                   [Enable internal runtime support and provide mpiexec/mpirun (default: enabled)])])
 
-    AC_ARG_WITH([prrte-platform],
-                [AC_HELP_STRING([--with-prrte-platform],
-                                [Platform file to use when building the internal PRRTE runtime support])])
+    # TODO: BWB: Check with Ralph if we need seperate files for OMPI and PRTE (which seems annoying)
+dnl    AC_ARG_WITH([prrte-platform],
```

Comment:
I made this option with only one 'r' - i.e., "prte"

Reply:
Is the project PRRTE or PRTE?  I'd like to be consistent through the whole system.  In terms of the actual argument, Open MPI will pass all arguments not explicitly called out down to PRTE.  So we don't actually have to do anything here; customers can do `./configure --with-platform=<foo> --with-prte-platform=<bar>` and the right things will happen.

But I should remove this code, since it isn't needed at this point.

Reply:
The project is "PRRTE", but the commands and variables are all with a single "r". I don't know how we want to expose the PRRTE and PMIx configure options at the user level for OMPI - I guess that is an open question? I grok that the right thing will happen - just not sure how the user can see what options exist.

Reply:
I'm not sure there's a great answer here, other than eventually getting `./configure --help=recursive` to work properly.  I should figure out why that doesn't.  Otherwise, we end up exposing way too many options (like why is `--with-platform` special compared to all the others?).

Reply:
I agree - I think we can set it aside for now as it may become too complex and confusing to the user. Perhaps all we need do at some point is add a comment to configure help indicating that these subprojects are included and direct them on how to find configure options specific to each subproject.

Reply:
Removed the commented out platform argument from the top level.  Also figured out how to make `./configure --help=recursive` work correctly, which I think is the answer to your question about how customers find arguments for sub-projects.